lib/vdc/BOVCollection.cpp

```diff
@@ -132,45 +145,64 @@ int BOVCollection::_parseHeader(std::ifstream &header)
             return _failureToReadError(DATA_FILE_TOKEN);
-        else if (rc == (int)parseCodes::FOUND)
+        else if (rc == (int)parseCodes::FOUND) {
             _dataFile = dataFile;
+            continue;
+        }
 
         double time;
         rc = _findToken(TIME_TOKEN, line, time);
         if (rc == (int)parseCodes::PARSE_ERROR)
             return _failureToReadError(TIME_TOKEN);
-        else if (rc == (int)parseCodes::FOUND)
+        else if (rc == (int)parseCodes::FOUND) {
             _time = time;
+            continue;
+        }
 
         std::string variable;
         rc = _findToken(VARIABLE_TOKEN, line, variable);
         if (rc == (int)parseCodes::PARSE_ERROR)
             return _invalidValueError(VARIABLE_TOKEN);
-        else if (rc == (int)parseCodes::FOUND)
+        else if (rc == (int)parseCodes::FOUND) {
             _variable = variable;
+            continue;
+        }
 
         rc = _findToken(GRID_SIZE_TOKEN, line, _tmpGridSize);
-        if (rc == (int)parseCodes::PARSE_ERROR) return _failureToReadError(GRID_SIZE_TOKEN);
+        if (rc == (int)parseCodes::PARSE_ERROR)
+            return _failureToReadError(GRID_SIZE_TOKEN);
+        else if (rc == (int)parseCodes::FOUND)
+            continue;
 
         rc = _findToken(FORMAT_TOKEN, line, _tmpDataFormat);
-        if (rc == (int)parseCodes::PARSE_ERROR) return _failureToReadError(FORMAT_TOKEN);
+        if (rc == (int)parseCodes::PARSE_ERROR)
+            return _failureToReadError(FORMAT_TOKEN);
+        else if (rc == (int)parseCodes::FOUND)
+            continue;
 
         // Optional tokens.  If their values are invalid, SetErrMsg, and return -1.
         //
         rc = _findToken(ORIGIN_TOKEN, line, _tmpBrickOrigin);
-        if (rc == (int)parseCodes::PARSE_ERROR) return _invalidValueError(ORIGIN_TOKEN);
+        if (rc == (int)parseCodes::PARSE_ERROR)
+            return _invalidValueError(ORIGIN_TOKEN);
+        else if (rc == (int)parseCodes::FOUND)
+            continue;
 
         rc = _findToken(BRICK_SIZE_TOKEN, line, _tmpBrickSize);
-        if (rc == (int)parseCodes::PARSE_ERROR) return _invalidValueError(BRICK_SIZE_TOKEN);
-
-        rc = _findToken(ENDIAN_TOKEN, line, _tmpDataEndian);
-        if (rc == (int)parseCodes::PARSE_ERROR) return _invalidValueError(ENDIAN_TOKEN);
+        if (rc == (int)parseCodes::PARSE_ERROR)
+            return _invalidValueError(BRICK_SIZE_TOKEN);
+        else if (rc == (int)parseCodes::FOUND)
+            continue;
 
         rc = _findToken(OFFSET_TOKEN, line, _tmpByteOffset);
-        if (rc == (int)parseCodes::PARSE_ERROR) return _invalidValueError(OFFSET_TOKEN);
+        if (rc == (int)parseCodes::PARSE_ERROR)
+            return _invalidValueError(OFFSET_TOKEN);
+        else if (rc == (int)parseCodes::FOUND)
+            continue;
 
         // All other variables are currently unused.
         //
+        _findToken(ENDIAN_TOKEN, line, _dataEndian);
```

Comment:
I thought endian stuff is removed?

Reply:
I'm still finding unused tokens, in case we want to implement them later.  Check out the lines below for other examples.